src/library.cpp

```diff
@@ -181,17 +186,16 @@ bool Library::removeBookmark(const std::string& zimId, const std::string& url)
 }
 
 
-void Library::dropReader(const std::string& id)
+void Library::dropCache(const std::string& id)
 {
-  mp_impl->m_readers.erase(id);
-  mp_impl->m_archives.erase(id);
+  mp_impl->mp_archiveCache->drop(id);
 }
 
 bool Library::removeBookById(const std::string& id)
 {
   std::lock_guard<std::mutex> lock(m_mutex);
   mp_impl->m_bookDB->delete_document("Q" + id);
-  dropReader(id);
+  dropCache(id);
   return mp_impl->m_books.erase(id) == 1;
 }
```

Comment:
Shouldn't the cache size be updated here too? If not, a comment must be added that it is not done intentionally

Reply:
Yes, I think it is not needed to update the cache size here. I've added a comment.